### PR TITLE
Feature/request transformer

### PR DESCRIPTION
**Pull request by italolelis**
## What does this PR do?

- Adds plugin to transform a request to an upstream. You can now modify headers and query string before the request is sent
- Removes viper when reading API definitions and OAuth Server files
- Adds structures to plugins configurations

### CODE REVIEW

File: pkg/loader/api_loader.go [hunk 1]
```diff
@@ -54,6 +54,10 @@ func (m *APILoader) RegisterAPI(referenceSpec *api.Spec) {
 			if pDefinition.Enabled {
 				logger.WithField("name", pDefinition.Name).Debug("Plugin enabled")
 				if p := m.pluginLoader.Get(pDefinition.Name); p != nil {
+					if pDefinition.Config == nil {
```
Comment (rafaeljesus) [inline pkg/loader/api_loader.go]: Do we need this even using Config as `map[string]interface{}`?

**Comment by vgarvardt** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/hellofresh/api-gateway/pull-request/150/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/api-gateway) [Approved with PullApprove](https://pullapprove.com/hellofresh/api-gateway/pull-request/150/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/api-gateway)

**Comment by rafaeljesus** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/hellofresh/api-gateway/pull-request/150/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/api-gateway) [Approved with PullApprove](https://pullapprove.com/hellofresh/api-gateway/pull-request/150/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/api-gateway)

**Review by rafaeljesus — APPROVED** [review]

**Comment by italolelis** [comment]
👍

File: pkg/middleware/request_transformer.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+// If and only if the header is already set, replace its old value with the new one. Ignored if the header is not already set.
+func (h *RequestTransformer) replaceHeaders(r *http.Request) headerFn {
+	return func(headerName string, headerValue string) {
+		if r.Header.Get(headerName) != "" {
+			r.Header.Set(headerName, headerValue)
+		}
+	}
+}
+
+func (h *RequestTransformer) trasnform(values map[string]string, fn headerFn) {
```
Comment (kieranajp) [inline pkg/middleware/request_transformer.go:89]: :| @italolelis @vgarvardt

Comment (rafaeljesus) [reply pkg/middleware/request_transformer.go:89]: English mafia :see_no_evil:

Comment (rafaeljesus) [reply pkg/middleware/request_transformer.go:89]: golint helps with misspell

**Pull request merged**